src/idom/utils.py

```diff
@@ -63,80 +67,149 @@ def html_to_vdom(source: str, *transforms: _ModelTransform) -> Dict[str, Any]:
-            if isinstance(style, str):
-                style_dict = {}
-                for k, v in (part.split(":", 1) for part in style.split(";") if part):
-                    title_case_key = k.title().replace("-", "")
-                    camel_case_key = title_case_key[:1].lower() + title_case_key[1:]
-                    style_dict[camel_case_key] = v
-                attrs["style"] = style_dict
-        return {"tagName": tag, "attributes": attrs, "children": []}
+    # Get rid of empty VDOM fields
+    _prune_vdom_fields(vdom)
+
+    return vdom
+
+
+def _mutate_vdom(vdom: Dict):
+    """Performs any necessary mutations on the VDOM attributes to meet VDOM spec.
+
+    Currently, this function only transforms the ``style`` attribute into a dictionary whose keys are
+    camelCase so as to be renderable by React.
+
+    This function may be extended in the future.
+    """
+    # Determine if the style attribute needs to be converted to a dict
+    if (
+        "attributes" in vdom
+        and "style" in vdom["attributes"]
+        and isinstance(vdom["attributes"]["style"], str)
+    ):
+        # Convert style attribute from str -> dict with camelCase keys
+        vdom["attributes"]["style"] = {
+            _hypen_to_camel_case(key.strip()): value.strip()
+            for key, value in (
+                part.split(":", 1)
+                for part in vdom["attributes"]["style"].split(";")
+                if ":" in part
+            )
+        }
+
+
+def _prune_vdom_fields(vdom: Dict):
+    """Remove unneeded fields from VDOM dict."""
+    if "children" in vdom and not len(vdom["children"]):
+        del vdom["children"]
+    if "attributes" in vdom and not len(vdom["attributes"]):
+        del vdom["attributes"]
+    if "key" in vdom and not vdom["key"]:
+        del vdom["key"]
+
+
+def _generate_vdom_children(
+    node: etree._Element, transforms: Iterable[_ModelTransform]
+) -> List[Union[Dict, str]]:
+    """Generates a list of VDOM children from an lxml node.
+
+    Inserts inner text and/or tail text inbetween VDOM children, if necessary.
+    """
+    return (  # Get the inner text of the current node
+        [node.text] if node.text else []
+    ) + list(
+        chain(
+            *(
+                # Recursively convert each child node to VDOM
+                [_etree_to_vdom(child, transforms)]
+                # Insert the tail text between each child node
+                + ([child.tail] if child.tail else [])
+                for child in node.iterchildren(None)
+            )
+        )
+    )
+
+
+def _hypen_to_camel_case(css_key: str) -> str:
+    """Convert a hypenated string to camelCase."""
+    first_word, *subsequent_words = css_key.split("-")
+
+    return "".join(
+        [
+            # Lowercase the first word
+            first_word.lower(),
+            # Use map() to titlecase all subsequent words
+            *map(str.title, subsequent_words),
+        ]
+    )
```

Comment:
```suggestion
def _hypen_to_camel_case(string: str) -> str:
    """Convert a hypenated string to camelCase."""
    first, remainder = string.split("-", 1)
    return first.lower() + remainder.title().replace("-", "")
```

Reply:
This causes an exception for CSS variables without a dash such as `color` or `background`.

Reply:
Good point. This will work though:
```suggestion
def _hypen_to_camel_case(string: str) -> str:
    """Convert a hypenated string to camelCase."""
    first, _, remainder = string.partition("-")
    return first.lower() + remainder.title().replace("-", "")
```